application/views/collection_instrument_view.py

```diff
@@ -17,9 +17,19 @@
 
 @auth.login_required
 @collection_instrument_view.route('/upload/<exercise_id>/<ru_ref>', methods=['POST'])
-def upload_collection_instrument(exercise_id, ru_ref):
+def upload_collection_instrument_with_ru(exercise_id, ru_ref):
     file = request.files['file']
-    msg = CollectionInstrument().upload_instrument(exercise_id, ru_ref, file)
+    classifiers = request.args.get('classifiers')
+    msg = CollectionInstrument().upload_instrument(exercise_id, file, ru_ref=ru_ref, classifiers=classifiers)
+    return make_response(msg, 200)
+
+
+@auth.login_required
+@collection_instrument_view.route('/upload/<exercise_id>', methods=['POST'])
+def upload_collection_instrument(exercise_id):
```

Comment:
I think you could write these two endpoints as one function like
```
@auth.login_required
@collection_instrument_view.route('/upload/<exercise_id>', methods=['POST'])
@collection_instrument_view.route('/upload/<exercise_id>/<ru_ref>', methods=['POST'])
def upload_collection_instrument(exercise_id, ru_ref=None):
    file = request.files['file']
    classifiers = request.args.get('classifiers')
    msg = CollectionInstrument().upload_instrument(exercise_id, file, ru_ref=ru_ref, classifiers=classifiers)
    return make_response(msg, 200)
```

I'm not sure if that's any cleaner than what's already there though